js/menu.js

```diff
@@ -2,13 +2,36 @@ var $menuElement = $('[data-name="Bottom icon bar"]');
 var menuInstanceId = $menuElement.data('id');
 var pageId = Fliplet.Env.get('pageId');
 
-if (menuInstanceId) {
-  init();
-}
-
 function init() {
-  $('body').addClass('fl-menu-bottom-bar');
   var data = Fliplet.Widget.getData(menuInstanceId) || {};
+  $('body').addClass('fl-menu-bottom-bar');
+
+  // Add exit app link
+  Fliplet.Hooks.on('addExitAppMenuLink', function () {
+    if ($menuElement.find('li[data-fl-navigate*="exit-app"]').length) {
+      return;
+    }
+
+    var $li = $([
+      '<li class="linked" data-fl-exit-app>',
+        '<div class="fl-bottom-bar-icon-holder">',
+          '<div class="fl-menu-icon">',
+            '<i class="fa fa-sign-out"></i>',
+          '</div>',
+          '<div class="fl-menu-title">',
+            '<span>Exit app</span>',
+          '</div>',
+        '</div>',
+      '</li>'
+    ].join(''));
+    $li.on('click', function onExitClick() {
+      Fliplet.Navigate.exitApp();
+    });
+    $(this).find('ul').append($li);
+
+    // Prevent default "Exit app" link from being added
+    return Promise.reject('');
```

Comment:
@tonytlwu you should reject with a truth-y error

Reply:
@squallstar Is that the convention? I followed the Form Builder method of rejecting with an empty string to stop further actions. I believe that's because Form Builder uses the reject message as error message.

Reply:
@tonytlwu that was just a hack to avoid the form builder from displaying an error messages, yes. Ideally I would prefer rejecting with either undefined (so no arguments) or a specific error like `new Error('ManuallyHandled')`, e.g. something we can parse and understand.

Reply:
As a further note, FormBuilder should be updated not to attempt displaying an error when undefined is given. We initially sent an empty string so it wouldn't fail when parsing it.

Reply:
OK. I'll reject with no arguments then. I'll leave the form builder changes to you when you next work on the component.